chef/automate
# Split Applications Load Generator into Separate Package

### :nut_and_bolt: Description: What code changed, and why?

Splits the `applications-load-gen` command into its own package. This will allow the load generator to be run as a habitat service, which will make it easier to run the load generator against our dev/acceptance instances.

### :chains: Related Resources

### :+1: Definition of Done

Can run the load generator as a separate service in the hab studio. We'll want it published to the hab depot and promoted past dev before integrating with the pipeline a2 instances.

### :athletic_shoe: How to Build and Test the Change

* `start_all_services`
* `build components/applications-load-gen`
* `mkdir -p /hab/user/applications-load-gen/config`
* Make a `/hab/user/applications-load-gen/config/user.toml` file like this except you need a working token for your automate:

```
[mlsa]
accept = true

[target]
host = "localhost"
auth_token = "BtyJlfl8RehYPx-A_7j5yA0DKkQ="
svc_count = 1000
```

* ` chef-automate dev stop-converge` (so the deployment service won't unload non-a2 services)
* `hab svc load YOUR_ORIGIN/applications-load-gen`, then `sl` to check for errors.
* View the applications UI and you should have 12 service groups. The load generator will change some services health checks every 5 minutes.
